gallery/src/components/NxCombobox/NxComboboxPredeterminedListExample.tsx

```diff
@@ -0,0 +1,53 @@
+/*
+ * Copyright (c) 2019-present Sonatype, Inc.
+ * This program and the accompanying materials are made available under
+ * the terms of the Eclipse Public License 2.0 which accompanies this
+ * distribution and is available at https://www.eclipse.org/legal/epl-2.0/.
+ */
+import React, { useCallback, useState } from 'react';
+import { filter, map, prepend, range } from 'ramda';
+import { NxCombobox, DataItem }
+  from '@sonatype/react-shared-components';
+
+const states:string[] = ['Alabama', 'Alaska', 'Arizona', 'Arkansas', 'California', 'Colorado', 'Connecticut',
+  'Delaware', 'Florida', 'Georgia', 'Hawaii', 'Idaho', 'Illinois', 'Indiana', 'Iowa', 'Kansas', 'Kentucky',
+  'Maine', 'Maryland', 'Massachusetts', 'Michigan', 'Minnesota', 'Mississippi', 'Missouri', 'Montana', 'Nebraska',
+  'Nevada', 'New Hampshire', 'New Jersey', 'New Mexico', 'New York', 'North Carolina', 'North Dakota', 'Ohio',
+  'Oklahoma', 'Oregon', 'Pennsylvania', 'Rhode Island', 'South Carolina', 'South Dakota', 'Tennessee', 'Texas', 'Utah',
+  'Vermont', 'Virginia', 'Washington', 'Washington DC', 'West Virginia', 'Wisconsin', 'Wyoming'];
+
+const items = prepend(
+    { id: 0, displayName: 'Loooooooooooooooooooooooooong Name' },
+    map(i => ({ id: i, displayName: states[i - 1] }), range(1, states.length + 1)));
+
+function search(query: string):DataItem<number, string>[] {
+  const lowercaseQuery = query.toLowerCase(),
+      matchingItems = filter(i => i.displayName.toLowerCase().includes(lowercaseQuery), items);
```

Comment:
Agree with the suggestion https://github.com/sonatype/sonatype-react-shared-components/pull/751#discussion_r964915382, Updated here c88f1abdb10667054bc42ea3d8ef210e6d81eaf1.

Reply:
> Agree with the suggestion https://github.com/sonatype/sonatype-react-shared-components/pull/751#discussion_r964915382, Updated here https://github.com/sonatype/sonatype-react-shared-components/commit/c88f1abdb10667054bc42ea3d8ef210e6d81eaf1.

At least one of the examples, the predetermined list example, still uses shows any result that contains the query, rather than only those that start with it

Reply:
Ah I think I misunderstood you when I saw the example link, thought the search logics should be applied to `both` only, but it should be applied to `both` and `list` instead, here is the updates e41403051a59c97f9102742fe0bd70f3c7941669.